code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm

```diff
@@ -0,0 +1,1195 @@
+	// Use the hidden . var to avoid needing to create a new local var (saves CPU)
+	. = list()
+	// Add everyone nearby to the list if they're in affinity, with key of the mob and value of the affinity
+	// . becomes list(jane = 1, tim = -3) etc
+	for(var/mob/living/M in people_nearby)
+		var/their_affinity = affinity[M.real_name]
+		if(their_affinity)
+			if(their_affinity >= 25 || their_affinity <= -10)
+				.[M] = affinity[M.real_name]
+	// Sort the list (timsort default sort comperator is numeric ascending, so highest affinity will be last in the list)
+	sortTim(., associative = TRUE)
+
+/datum/ai_holder/simple_mob/teppi/handle_wander_movement()
+	var/mob/living/simple_mob/vore/alienanimals/teppi/tepholder = holder
+	if(tepholder.resting)
+		if(prob(5))
+			tepholder.lay_down()
+		return
+	// Copypasta from parent handle_wander_movement
+	if(isturf(holder.loc) && can_act())
+		if(--wander_delay > 0)
+			return
+		if(!wander_when_pulled && (holder.pulledby || holder.grabbed_by.len))
+			ai_log("handle_wander_movement() : Being pulled and cannot wander. Exiting.", AI_LOG_DEBUG)
+			return
+	// We're having our chance NOW
+	wander_delay = base_wander_delay
+	// Typecast the ai_holder 'holder' var as a teppi so we can call do_I_know_you()
+	var/list/affinity_nearby = tepholder.do_I_know_you()
+	var/turf/T // Turf we might eventually move to
+	// If we found any affinity people nearby
+	if(affinity_nearby.len)
+		// Extract the highest affinity person from the list, by taking the last item (the item at
+		// position 6 in a list that's 6 length is the last item eg)
+		var/mob/living/L = affinity_nearby[affinity_nearby.len]
+		// If >= 0, wander towards
+		if(affinity_nearby[L] >= 0)
+			T = get_step_to(holder, L, 1)
+		// Else wander away
+		else
+			T = get_step_away(holder, L)
+	// Didn't find affinity people nearby, copypasta from normal wandering.
+	// We don't call ..() because it'll perform some of the same work again and want to avoid that
+	if(!T)
+		if(prob(5))
+			tepholder.lay_down()
+			return
+		var/moving_to = 0 // Apparently this is required or it always picks 4, according to the previous developer for simplemob AI.
+		moving_to = pick(cardinal)
+		holder.set_dir(moving_to)
+		T = get_step(holder,moving_to)
+	// Finally do move if we actually found somewhere we'd like to go
+	if(T)
+		holder.IMove(T)
+
+/datum/ai_holder/simple_mob/teppi/handle_idle_speaking()
+	if(holder.resting)
+		return
+	..()
+
+/datum/ai_holder/simple_mob/teppi/baby/handle_idle_speaking()
+	if(holder.resting)
+		return
+	..()
+
+/datum/ai_holder/simple_mob/teppi/on_hear_say(mob/living/speaker, message)
+	var/mob/living/simple_mob/vore/alienanimals/teppi/T = holder
+	if(holder.client)
+		return
+	if(!speaker.client)
+		return
+	if(!T.teppi_adult)
+		return
+	var/speaker_affinity = T.affinity[speaker.real_name]
+	if(findtext(message, "lets go") || findtext(message, "come teppi") || findtext(message, "come [holder.name]"))
```

Comment:
Comment: You could probably fix the apostrophe problem by doing message = `html_decode(message)` (or maybe it's urlencoded, I dunno so in that case url_decode(message))

Reply:
This worked with html_decode I think!

Reply:
\o/